src/scenic/core/pruning.py

```diff
@@ -380,79 +396,125 @@ def pruneVisibility(scenario, verbosity):
     for obj in scenario.objects:
         # Extract the base region if it exists
         position = currentPropValue(obj, "position")
-        base, offset = matchInRegion(position)
+        base, offset, pir_dist = matchInRegion(position)
 
-        if base is None or needsSampling(base):
+        # Compute the maximum distance the object can be from the sampled point
+        if offset is not None:
+            _, maxDistance = supportInterval(offset.norm())
+        else:
+            maxDistance = 0
+
+        if (
+            base is None
+            or needsSampling(base)
+            or needsSampling(obj.radius)
+            or maxDistance is None
+        ):
             continue
 
-        newBase = base
+        currBase = base
+        currDist = pir_dist
+        currPos = position
 
-        # Define a helper function to buffer an oberver's visibleRegion, resulting
+        # Define a helper function to attempt buffer an oberver's visibleRegion, resulting
         # in a region that contains all points that could feasibly be the position
-        # of obj, if it is visible from the observer.
+        # of obj, if it is visible from the observer. If possible buffer exactly, otherwise
+        # try to buffer approximately, and if that is also not feasible just return the viewRegion.
         def bufferHelper(viewRegion):
-            return viewRegion._bufferOverapproximate(obj.radius / 2, PRUNING_PITCH)
+            buffer_quantity = obj.radius / 2 + maxDistance
+            if hasattr(viewRegion, "buffer"):
+                return viewRegion.buffer(buffer_quantity)
+            elif hasattr(viewRegion, "_bufferOverapproximate"):
+                if needsSampling(viewRegion):
+                    return viewRegion._bufferOverapproximate(buffer_quantity, 1)
+                else:
+                    current_pitch = PRUNING_PITCH
+                    buffered_container = None
+
+                    while buffered_container is None:
+                        buffered_container = viewRegion._bufferOverapproximate(
+                            buffer_quantity, current_pitch
+                        )
+
+                        if isinstance(buffered_container, VoxelRegion):
+                            buffered_container = buffered_container.mesh
+
+                        current_pitch = min(2 * current_pitch, 1)
+
+                    assert buffered_container is not None
+
+                    return buffered_container
+            else:
+                return viewRegion
```

Comment:
Isn't this case unsound? If we intersect with the raw view region we could lose some valid positions where the object would be visible. If we really cannot buffer the view region we should not attempt to prune.

Reply:
Good catch! I copied this over from the erosion function since we can just do a best effort there. I'll remove it and add an assertion, but the good news is that this should never happen (since we have the approximate fast path).